### ISSUE TITLE
please help me to fix this bugs

### ISSUE THREAD

**Issue by pasinduc** [koalaman/shellcheck#2407]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2128, SC2128, SC2128, SC2128
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
port1=(4132 4133 4134 4135 4136)
port2=(3032 3033 3034 3035 3036)
username1=('gvsfans1' 'gvsfans2' 'gvsfans3')
username4=('gvsfans1' 'gvsfans2' 'gvsfans3')
for i in ${!username1[*]}; do


echo "$username1"["$i"]
echo "$username4"["$i"]
echo "$port1"["$i"]
echo "$port2"["$i"]
done
```

#### Here's what shellcheck currently says:
```
Line 9	SC2128: Expanding an array without an index only gives the first element.
Line 10	SC2128: Expanding an array without an index only gives the first element.
Line 11	SC2128: Expanding an array without an index only gives the first element.
Line 12	SC2128: Expanding an array without an index only gives the first element.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by brother** [comment]
```shell
#!/bin/bash
port1=(4132 4133 4134 4135 4136)
port2=(3032 3033 3034 3035 3036)
username1=('gvsfans1' 'gvsfans2' 'gvsfans3')
username4=('gvsfans1' 'gvsfans2' 'gvsfans3')
for i in ${!username1[*]}; do


echo "${username1["$i"]}"
echo "${username4["$i"]}"
echo "${port1["$i"]}"
echo "${port2["$i"]}"
done
```
